SCClassLibrary/Common/Core/Kernel.sc

```diff
@@ -452,11 +462,28 @@ FunctionDef {
 		_FunctionDefDumpContexts
 		^this.primitiveFailed
 	}
+
+	// These methods are used to indicate that only a subsection of the argName
+	//     and prototypeFrame are used to call the FunctionDef.
+	argNamesForCallStartIndex { ^0 }
+	argNamesForCallSize { ^argNames.size }
+
+	argNamesForCall {
+		^argNames
+		.drop(this.argNamesForCallStartIndex)
+		.keep(this.argNamesForCallSize)
+	}
+
+	defaultArgsForCall {
+		^prototypeFrame
+		.drop(this.argNamesForCallStartIndex)
+		.keep(this.argNamesForCallSize)
+	}
+
 	inspectorClass { ^FunctionDefInspector }
 
-	findReferences { arg aSymbol, references;
-		var lits;
-		lits = selectors.asArray;
+	findReferences { |aSymbol, references|
```

Comment:
It would be good to minimise reformatting please for now, not to muddy the water. It is a critical PR and in the end, we will have to be able to know what was changed. 

Otherwise, I also would like to have these cosmetics, but right now it feels like noise.